packages/terra-list/src/SelectableList.jsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import List from './List';
+import SelectableUtils from './SelectableUtils';
+
+const propTypes = {
+  /**
+   * The children list items passed to the component.
+   */
+  children: PropTypes.node,
+  /**
+   * Whether or not the child list items should have a border color applied.
+   */
+  isDivided: PropTypes.bool,
+  /**
+   * Whether or not unselected items should be disabled.
+   */
+  disableUnselectedItems: PropTypes.bool,
+  /**
+   * Whether or not the child list items has a disclosure indicator presented.
+   * The behavior is intended to be used with a single selection style list, as multi selection style list should not perform disclosures.
+   */
+  hasChevrons: PropTypes.bool,
+  /**
+   * A callback event that will be triggered when selection state changes.
+   */
+  onChange: PropTypes.func,
+  /**
+   * An array of the currectly selected indexes.
+   */
+  selectedIndexes: PropTypes.array,
```

Comment:
To match the existing API I think this makes sense.

In the future it might be worth discussing if selectedKeys might be a more appropriate identifier of selected items. Each List item should be provided a unique key ( ideally not index ). The key would always map to the item, whereas an index has no constraints or requirements to always uniquely identify an item but a key by definition **should**. It would help remove the awkwardness of remapping indexes to items after any additions / removals from the list.

Reply:
We could change this under the hood. And just have the current multi and single selects return indexes. Since this is a new subcomponent, we could start the pattern here.

Reply:
After some looking, react doesn't allow you to use the key. It's considered protected and not stored on the child. We'd need special item key props or meta data to accomplish this.

Reply:
The key is not accessible within props, but it should still be accessible on the child. If this is causing too many changes I wouldn't make the changes on this pull request. 

Here is a small code sample I use that accesses the key in a similar way: 
```
    const children = React.Children.map(this.props.children, child => (
      <SelectableItem onClick={() => this.handleSelection(child.key)} isSelected={child.key === this.state.selectedKey}>
        {child}
      </SelectableItem>
    ));
```

https://github.com/facebook/react/issues/2429#issuecomment-277810636

Reply:
I got this working with the keys, but after some discussion, maintaining both the index and key for passivity is too complicated for now. So we'll just handled indexes in this PR.